config/config.go

```diff
@@ -94,6 +94,7 @@ type configFile struct {
 }
 
 type keyGroup struct {
+	Merge   []keyGroup
 	KMS     []kmsKey
```

Comment:
I'm wondering whether you can create an infinite loop by using YAML anchors. I don't think you can, but YAML anchors are a feature of YAML I try to avoid, so maybe I'm missing a trick here.

Reply:
I also don't think it is possible, because YAML anchors only resolve references when parsed (and the parser supports it). But I will double check this.

Reply:
Tested:
```
creation_rules:
  - path_regex: ""
    key_groups:
    - merge: &merge1
      - merge: *merge1
```
Result:
```
Expected nil, but got: &errors.errorString{s:"Could not unmarshal config file: yaml: anchor 'merge1' value contains itself"}
```
So I think it is safe to say that an infinite loop like this could not happen.